static/app/views/dashboardsV2/utils.tsx

```diff
@@ -43,3 +45,42 @@ export function eventViewFromWidget(
     environment: environments,
   });
 }
+
+function coerceStringToArray(value?: string | string[] | null) {
+  return typeof value === 'string' ? [value] : value;
+}
+
+export function constructWidgetFromQuery(query: Query): Widget | undefined {
+  if (query) {
```

Comment:
Typing here would suggest that query should always exist? If you think it can be undefined I'd change types to reflect that, otherwise the if probably isn't necessary.

Reply:
Updated type to allow for undefined!